stages/org.osbuild.systemd-journald

```diff
@@ -0,0 +1,113 @@
+
+The `config` option allows to create a systemd-journald configuration drop-in
+file in `/usr/lib/systemd/journald.conf.d` with the name `filename`.
+
+Drop-in configuration files can specify the following subset of options:
+- 'Journal' section
+  -  'Storage' option
+  -  'Compress' option
+  -  'SplitMode' option
+  -  'MaxFileSec' option
+  -  'MaxRetentionSec' option
+  -  'SyncIntervalSec' option
+  -  'Audit' option
+
+At least one option must be specified in the 'Journal' section.
+"""
+
+import configparser
+import os
+import sys
+
+import osbuild.api
+
+SCHEMA = r"""
+"additionalProperties": false,
+"required": ["filename", "config"],
+"properties": {
+    "filename": {
+        "type": "string",
+        "description": "Name of the systemd-journald drop-in file",
+        "pattern": "^[a-zA-Z0-9_\\.-]{1,250}\\.conf$"
+    },
+    "config": {
+        "additionalProperties": false,
+        "type": "object",
+        "description": "systemd-journald configuration",
+        "minProperties": 1,
+        "properties": {
+            "Journal": {
+                "additionalProperties": false,
+                "type": "object",
+                "description": "'Journal' configuration section",
+                "minProperties": 1,
+                "properties": {
+                    "Storage": {
+                        "type": "string",
+                        "description": "Controls where to store journal data.",
+                        "pattern": "^(volatile|persistent|auto|none)$"
+                    },
+                    "Compress": {
+                        "type": "string",
+                        "description": "Sets whether the data objects stored in the journal should be compressed or not. Can also take threshold values."
+                    },
+                    "SplitMode": {
+                        "type": "string",
+                        "description": "Splits journal files per user or to a single file.",
+                        "pattern": "^(uid|none)$"
+                    },
+                    "MaxFileSec": {
+                        "type": "string",
+                        "description": "Max time to store entries in a single file. By default seconds, may be sufixed with units to override this."
+                    },
+                    "MaxRetentionSec": {
+                        "type": "string",
+                        "description": "Maximum time to store journal entries."
+                    },
+                    "SyncIntervalSec": {
+                        "type": "string",
+                        "description": "Timeout before synchronizing journal files to disk."
+                    },
+                    "Audit": {
+                        "type": "string",
+                        "description": "Enables/Disables kernel auditing on start-up, leaves it as is if unspecified.",
+                        "pattern": "^(yes|no){0,1}$"
+                    }
+                }
```

Comment:
I think it would be easier to read if all the patterns were replaced with `"enum"`.  For example the first one would be:
```json
"Storage": {
  "type": "string",
    "description": "Controls where to store journal data.",
    "enum": ["volatile", "persistent", "auto", "none"]
},
```

In the case of `"Audit"`, would be make more sense to have it as a boolean and write it out as yes/no accordingly?  Otherwise, an enum would work better here as well.

Reply:
`Audit` is no boolean, but a tristate. But I agree, an enum might work best. Something like `["yes", "no", "default"]`?

Reply:
I think that the third state in `Audit` is leaving the option specified, but unset, so from my point of view the best option to handle this is the regex pattern.

Will update the other options :+1:

Reply:
@achilleas-k @dvdhrm so, what do we do with the `Audit` option? If we use an enum with `["yes", "no", "default"]` we will have to substitute that `default` with an empty string (the option has to be set unset), and if we choose the regex with the pattern that that would handle it.

Reply:
That sounds good :+1: , I'll update it in a second